yellowbrick/text/postag.py

```diff
@@ -139,16 +144,34 @@ def fit(self, X, y=None, **kwargs):
         self : instance
             Returns the instance of the transformer/visualizer
         """
-        # TODO: add support for other tagsets?
-        if self.tagset == "penn_treebank":
-            self.pos_tag_counts_ = self._penn_tag_map()
-            self._handle_treebank(X)
-
-        elif self.tagset == "universal":
-            self.pos_tag_counts_ = self._uni_tag_map()
-            self._handle_universal(X)
+        if(self.stack == True and y == None):
+            raise YellowbrickValueError("Specify y for stack=True")
+        
+        labeled = []
+        if(y!=None):
+            for i in range(len(y)):
+                labeled.append((X[0][i],y[i]))
+        self.label_ = set(map(lambda x:x[1], labeled))
+        
+        if(self.stack == True):
+            X = [[[y[0]] for y in labeled if y[1]==x] for x in self.label_]
+        else:
+            X = [X]
+        
+        self.last = None
+        self.stack_count = 0
```

Comment:
Lines 150-162 are an interesting solution to the problem of how to count part of speech tags by class. I especially like the novel idea of drawing each bar one at a time, growing the bar size on each pass because the `self._handle` methods are cumulative. 

Unfortunately, this does bring up an issue - and that is that there is now a slight mismatch between the visual stacks and the data that the user might introspect, namely `self.pos_tag_counts_` -- which only represents the total chart and not the different stacks. One thing that we want to be conscious of is that the computations we make for the visualization might be relevant to the user for diagnostic purposes. 

I propose that if the user wanted a break down of the part of speech distribution by class (e.g. by passing `stack=True`) then they would also want to be able to access that data in `self.pos_tag_counts_`. E.g. these two cases:

```python
# stack is False 
self.pos_tag_counts_ == {
    "NN": 42, 
    "NNP": 12,
    "VBG: 27, 
    ...
}

# stack is True 
self.pos_tag_counts_ == {
     "books": {  
        "NN": 42, 
        "NNP": 12,
        "VBG: 27, 
        ...
    },
    "cooking": {  
        "NN": 112, 
        "NNP": 72,
        "VBG: 249, 
        ...
    },
}
```

The amount of work you're doing in `fit()` could be minimized by computing the unique labels in `y` using `np.unique` (otherwise `None`), which `self._*_tag_map()` would take advantage of if present. The `self._handle_*` methods could also take `y` as input, and they could construct the correct learned output as above. Then draw would only have to know about `self.pos_tag_counts_` as before without having to track the stack count or the last stack marker. 

Although a relatively small change, I believe this would have a very big impact on the usability of the visualizer. Would you be willing to try to make the changes I suggested above?

Reply:
Hello @bbengfort, I will love to make these changes. I will get back to you by tomorrow with all the amendments.
My interpretation of the above is as follows. You suggest making a new function _*_tag_map() which returns a dictionary having the pos_tag_count for each label(if stack is True). And then use this returned dictionary in draw() to generate the stacked barplot. 
Is this what you mean? If not then please let me know about what needs to be done!

Reply:
Hi Naresh, please take your time - there is no rush! 

And thank you for asking for clarification! I do not mean to create a new function, I'm simply using a shorthand to identify both `_uni_tag_map` and `_penn_tag_map` - same with the `_handle_treebank` and `_handle_universal` methods. 

The `_uni_tag_map` and `_pen_tag_map` should return a different datastructure depending on `self.stack` and `self.labels_`, which is saved on `self.pos_tag_counts_`. 

`_handle_treebank` and `_handle_universal` will have to accept an addtional, optional `y=None` argument, and will modify `self.pos_tag_counts_` accordingly. Then `draw()` will use `self.pos_tag_counts_`.

Reply:
Thanks!!